### PR TITLE
Migrate OOB flow to localhost flow for OAuth2 auth

### LINKED ISSUE #2378: Migrate OOB flow to localhost flow for OAuth2 authentication
**Describe the bug**
Google deprecated the OOB Oauth2 flow. The Timesketch API client needs to start using the localhost http server flow to authenticate via OAuth2.

**To Reproduce**
1. Clear cached credentials
2. Attempt to use the command-line timesketch cli client and authenticate via Google OAuth2

**Expected behavior**
Authentication should be successful.

### PR DESCRIPTION
**Pull request by jleaniz**
Please provide enough information so that others can review your pull request:

This PR addresses issue #2378.

**Checks**

- [X] All tests succeed.
- [X] Unit tests added.
- [X] e2e tests added.
- [X] Documentation updated.

**Closing issues**
closes #2378

### CODE REVIEW

File: api_client/python/timesketch_api_client/client.py [hunk 1]
```diff
@@ -247,7 +248,7 @@ def _create_oauth_session(
             flow.redirect_uri = self.DEFAULT_OAUTH_OOB_URL
 
         if run_server:
-            _ = flow.run_local_server()
+            _ = flow.run_local_server(host="localhost", port=8080, open_browser=False)
```
Comment (berggren) [inline api_client/python/timesketch_api_client/client.py:251]: Please add host, port, open_browser as parameter to this function, with these values as defaults. Then anyone using the api client can choose.

**Review by berggren — CHANGES_REQUESTED** [review]
Looks god, one small request.

**Comment by berggren** [comment]
Can you also up the version in: https://github.com/google/timesketch/blob/master/api_client/python/timesketch_api_client/version.py

Then we can built a new pypi package

File: api_client/python/timesketch_api_client/client.py (hunk 1 above)
Comment (jleaniz) [reply api_client/python/timesketch_api_client/client.py]: Done

**Comment by jleaniz** [comment]
Updated the API version

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**